### DISCUSSION TITLE
OpenFOAM-ADF installation problem

### DISCUSSION THREAD

**Discussion by ryupuriwat** [mdolab/dafoam#628] [Installation]
Now, I am installing OpenFOAM OpenFOAM-ADR OpenFOAM-ADF according to the tutorial. I would like to know what is the reason that  OpenFOAM and OpenFOAM-ADR can be installed sucessfully but only  OpenFOAM-ADF have a problem as there is no folder in side OpenFOAM-v1812-ADF/platforms/linuxARM64GccDPInt32Opt/bin after installation so when I do source bash I can not do it sucessfully.
![0a48faab75ec5bb0c65fb319c13e1cc](https://github.com/mdolab/dafoam/assets/109540234/25508ea6-2443-4e1b-9456-57f763842e31)
what can I notice is that the installation of OpenFOAM-ADR OpenFOAM-ADF use the same folder to install just a bit of modification. Since, OpenFOAM-ADR can be install sucessfully OpenFOAM-ADF also should be. Moreover, when I check into the Allwmake, I notice that "$WM_CODI_AD_MODE" = "CODI_AD_REVERSE"  is inside the Allwmake even though it is for OpenFOAM-ADF. So right now I am stucking in this step. Could you please tell me what is the possible problem because I follow all step in the official website for installation and stuck at it for many days and still can not fix it. 
![image](https://github.com/mdolab/dafoam/assets/109540234/c4ce9a52-76e9-4050-9eda-c982c364b250)

**Comment by friedenhe** [comment]
What you saw is normal. Your ADF compilation is good. The reason we did not see the bin folder is that the ADF version is designed to not have any exes in the bin folder. Note that the DASimpleFoamReverseAD for the ADR version is used only for regression tests. This exe is never used in optimization.
